### DISCUSSION TITLE
Cannot pass array as args in useContractRead

### DISCUSSION THREAD

**Discussion by 1337-ishaan** [wevm/wagmi#3055] [Q&A] [answered]
This is my ABI object for getting total prices 

 
```
   {
      inputs: [
        { internalType: "uint256[]", name: "coordinateIds", type: "uint256[]" },
      ],
      name: "getTotalPrice",
      outputs: [{ internalType: "uint256", name: "", type: "uint256" }],
      stateMutability: "view",
      type: "function",
    },
```

I want to use `wagmi` hook `useContractReads` for passing the args and getting the total prices

```
const { data, isError, isLoading } = useContractReads({
    contracts: [
      {
        address: coordinateGrid.contractAddress,
        abi: coordinateGrid.abi,
        functionName: "getTotalPrice",
        args: selectedCoordIds, // selectedCoordIds is a number[] | eg [23,212,342,554]
      },
    ],
  });

```

I get the following error
```
query.mjs:252 AbiEncodingLengthMismatchError: ABI encoding params/values length mismatch.
Expected length (params): 1
Given length (values): 4
```

```
[
    "This could be due to any of the following:",
    "  - The contract does not have the function \"getTotalPrice\",",
    "  - The parameters passed to the contract function may be invalid, or",
    "  - The address is not a contract.",
    " ",
    "Contract Call:",
    "  address:   0xF736Aa58E3861544448BF5735363614fbfCBC870\n  function:  getTotalPrice(uint256[] coordinateIds)\n  args:                   (100)"
]
```

I have verified that my contract address and ABI are of not different, is there something we need to pass in args?

useContractWrite function works, although I have the same args that are needed to pass there

ABI:
```
    {
        "inputs": [
            {
                "internalType": "uint256[]",
                "name": "coordinateIds",
                "type": "uint256[]"
            }
        ],
        "name": "purchaseCoordinates",
        "outputs": [],
        "stateMutability": "payable",
        "type": "function"
    }
```
Purchase wagmi Hook:

     const { config } = usePrepareContractWrite({
        address: coordinateGrid.contractAddress,
        abi: coordinateGrid.abi,
        functionName: "purchaseCoordinates",
        args: [selectedCoordIds],
      });
    
     const { write: purchaseCoordinates } = useContractWrite(config);

I tried passing `args:[selectedCoordIds]` but it shows the same error

**Comment by tmm** [comment, marked as answer]
Check out this [TypeScript Playground](https://www.typescriptlang.org/play?target=9#code/JYWwDg9gTgLgBAbzgVwM4FMDCEB2MoCGAxjAEroEAmqcAvnAGZQQhwDkA7gQOYjBsAoAUVyp4BAEbA4AXjgBtAXERLlcYDjDIYqAFwLVaxOrzooOAgBsAKgE8w6fW2QaYAJgCsANnkBdNgA0cBYgjuwi0JQaBDDoAJLUgXAw9mHOrp4+-nQBhnC+uUYhadzoMNYQMFYAClDAROiBeRDaWjr68kiuZhY2qU4ueJlJKQ4DGd5sdAV5YjHoALLaksCWwClOAG7A6BxNRqNpDMg4JMC4+zkCvnAENCI4YkIPYnBQ6KjIlvByaFi4+GIZAo1AAFAhVA9ASQ9AYjBCjMoqJR3qhYWwAAwAD0uRhWhURx1OMHOOAAcgRQk5SuVKjU6g1cWoCFBuLD5PIAIxuADMOCCABZvPy4AB2AAcAE4cL4Zoi4AB6BVwAB6AH48rQCTNaABKASor4wARAA).
